libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/hooks/useDependentDateFiltersConfiguration.ts

```diff
@@ -1,34 +1,98 @@
+import {
+    IDashboardDependentDateFilter,
+    setAttributeFilterDependentDateFilters,
+    useDispatchDashboardCommand,
+} from "../../../../../../model/index.js";
 import { useDependentDateFilterConfigurationState } from "./useDependentDateFilterConfigurationState.js";
 
 export function useDependentDateFiltersConfiguration(
     neighborDateFilters: IDashboardDateFilter[],
     currentFilter: IDashboardAttributeFilter,
     commonDateFilter?: IDashboardDateFilter,
 ) {
-    const { localIdentifier: currentFilterLocalId } = currentFilter.attributeFilter;
+    const { filterElementsByDate, localIdentifier: currentFilterLocalId } = currentFilter.attributeFilter;
 
     invariant(
         currentFilterLocalId,
         "Cannot initialize the attribute filter configuration panel, filter has missing 'localIdentifier' property",
     );
 
-    const originalState = useDependentDateFilterConfigurationState(neighborDateFilters, commonDateFilter);
+    const saveDependentDateFilterCommand = useDispatchDashboardCommand(
+        setAttributeFilterDependentDateFilters,
+    );
+
+    const originalState = useDependentDateFilterConfigurationState(
+        neighborDateFilters,
+        commonDateFilter,
+        filterElementsByDate,
+    );
 
     const [dependentDateFilters, setDependentDateFilters] =
         useState<IDashboardDependentDateFilter[]>(originalState);
 
+    const onDependentDateFiltersSelect = (dataSet: ObjRef, isSelected: boolean) => {
+        invariant(!isUriRef(dataSet));
+
+        const changedParentIndex = dependentDateFilters.findIndex(
+            (dependentDateFilter) => dependentDateFilter.localIdentifier === dataSet.identifier,
+        );
+        const changedItem = { ...dependentDateFilters[changedParentIndex] };
+
+        changedItem.isSelected = isSelected;
+
+        const changedParentItems = [...dependentDateFilters];
+        changedParentItems[changedParentIndex] = changedItem;
+
+        setDependentDateFilters(changedParentItems);
+    };
+
+    const onDependentDateFiltersConfigurationChanged = useMemo<boolean>(() => {
+        return !isEqual(dependentDateFilters, originalState);
+    }, [dependentDateFilters, originalState]);
+
     const onConfigurationClose = useCallback(() => {
         setDependentDateFilters(originalState);
     }, [originalState]);
 
+    const onDependentDateFiltersChange = useCallback(() => {
+        // dispatch the command only if the configuration changed
+        if (onDependentDateFiltersConfigurationChanged) {
+            const dateFilters: IDashboardAttributeFilterByDate[] = [];
+            dependentDateFilters.forEach((dependentDateFilter) => {
+                if (!dependentDateFilter.isSelected) {
+                    return;
+                }
+
+                dateFilters.push({
+                    filterLocalIdentifier: dependentDateFilter.localIdentifier,
+                });
+            });
```

Comment:
Highly optional but this is 3 lines vs 9 and is bit more readable (at least to me):

```
const dateFilters = dependentDateFilters
   .filter((filter) => filter.isSelected())
   .map((filter) => ({  filterLocalIdentifier: filter.localIdentifier }));
```